samples/nrf9160/pdn/src/main.c

```diff
@@ -25,6 +29,65 @@ static const char * const event_str[] = {
 	[PDN_EVENT_IPV6_DOWN] = "IPv6 down",
 };
 
+static void ifaddrs_print(void)
+{
+	int err;
+
+	struct nrf_ifaddrs *ifaddrs;
+	struct nrf_ifaddrs *ifa;
+
+	char addr[NRF_INET6_ADDRSTRLEN] = {0};
+	char netmask[NRF_INET6_ADDRSTRLEN] = {0};
+	char broadaddr[NRF_INET6_ADDRSTRLEN] = {0};
+	char dstaddr[NRF_INET6_ADDRSTRLEN] = {0};
+
+	err = nrf_getifaddrs(&ifaddrs);
+	if (err) {
+		printk("nrf_getifaddrs() failed, err %d\n", err);
+		return;
+	}
+
+	printk("\nInterface addresses:\n");
+
+	ifa = ifaddrs;
+	while (ifa != NULL) {
+		switch (ifa->ifa_netmask->sa_family) {
+		case NRF_AF_INET:
+			nrf_inet_ntop(ifa->ifa_addr->sa_family,
+				      &SIN(ifa->ifa_addr)->sin_addr,
+				      addr, sizeof(addr));
+			break;
+		case NRF_AF_INET6:
+			nrf_inet_ntop(ifa->ifa_addr->sa_family,
+				      &SIN6(ifa->ifa_addr)->sin6_addr,
+				      addr, sizeof(addr));
+			break;
+		default:
+			printk("Unknown address family, %d\n", ifa->ifa_netmask->sa_family);
+		}
+
+		nrf_inet_ntop(ifa->ifa_netmask->sa_family,
+			      &SIN(ifa->ifa_netmask)->sin_addr,
+			      netmask, sizeof(netmask));
+		nrf_inet_ntop(ifa->ifa_broadaddr->sa_family,
+			      &SIN(ifa->ifa_broadaddr)->sin_addr,
+			      broadaddr, sizeof(broadaddr));
+		nrf_inet_ntop(ifa->ifa_dstaddr->sa_family,
+			      &SIN(ifa->ifa_dstaddr)->sin_addr,
+			      dstaddr, sizeof(dstaddr));
+
+		printk("Address: %s\n", addr);
+		printk("Netmask: %s\n", netmask);
+		printk("Broadcast address: %s\n", broadaddr);
```

Comment:
As Broadcsat/Destination are a union, suggest only outputting one, along with the relevant flags:

```
printk("Broadcast/Destination [%d] address (%d): %s\n", ifa->ifa_flags, ifa->ifa_broadaddr->sa_family, broadaddr);
```

Reply:
If you want to resolve the flags:

```
   int BROADCAST = 1<<1;
   int POINTTOPOINT = 1<<4;
   if (ifa->ifa_flags && BROADCAST > 0) {
     ...
```